README.md

```diff
@@ -112,6 +111,45 @@ query {
   }
 }
 ```
+## Testing Guidelines
+When writing tests, try to follow these guidelines if possible:
+
++ Tests should be as readable as possible and not complex at all. You should understand them by looking at them just once.
++ Local helper functions defined in test files should have functional and easy-to-understand rather than technical names. Meaning, `clickNewUserButton()` is better than `clickElementByTypeAndTestId('button','new-user-button')`.
++ More general use helpers like 'clickElementByTypeAndTestId' can be used within the local helper functions if preferred. The reason for functional naming preference lies in increased readability of tests.
++ Avoid any duplication of helper functions across several files! If using the same functions in several tests (files), there's a tendency to copy-paste the whole file and then rewrite tests. This leads to code duplication of helper functions. Instead, helper functions needed in several locations should be defined in one place should be available globally. Find the matching one by name or create a new one. Avoid creating miscellaneuos file names as it tends to lead to chaos.
++ Current codebase doesn't 100% follow everything stated above but it'd definitely help organising the test helpers accordingly from now on.
+
+![Selection_599](https://user-images.githubusercontent.com/8964422/77221481-6a190d00-6b4a-11ea-88d7-9fc70ce1c982.png)
```

Comment:
Perhaps add some link resources on best practices for tests? I can dig some up if needed.

Reply:
These guidelines and the flow diagram actually came from Martin for the legacy project. I did not want to loose them and put them here. 

We can write more about Tests, but I would make it then plattform specific and put it in the different frontend and backend folders. Testing React and Flask is a bit different. 

If you have other general testing best practises, feel free to add.

Reply:
@HaGuesto General best practices: https://medium.com/better-programming/13-tips-for-writing-useful-unit-tests-ca20706b5368